File: backend/src/main/java/com/jobtracker/entity/Reminder.java
package com.jobtracker.entity;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Table(name = "reminders")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reminder {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "application_id")
    private JobApplication application;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "interview_id")
    private Interview interview;
    
    @Column(nullable = false)
    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String description;
    
    @Column(nullable = false)
    private LocalDate date;
    
    @Column
    private LocalTime time;
    
    @Column(name = "reminder_type", nullable = false)
    private String reminderType;
    
    @Column(name = "is_completed")
    private Boolean isCompleted = false;
    
    @Column(name = "is_recurring")
    private Boolean isRecurring = false;
    
    @Column(name = "recurring_pattern")
    private String recurringPattern;
    
    @Column(name = "notification_sent")
    private Boolean notificationSent = false;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    private LocalDateTime updatedAt;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: frontend/src/types/reminder.ts
export interface ReminderRequest {
  applicationId?: number;
  interviewId?: number;
  title: string;
  description?: string;
  date: string;
  time?: string;
  reminderType: string;
  isRecurring?: boolean;
  recurringPattern?: string;
}

export interface ReminderResponse {
  id: number;
  applicationId?: number;
  applicationPosition?: string;
  interviewId?: number;
  title: string;
  description?: string;
  date: string;
  time?: string;
  reminderType: string;
  isCompleted: boolean;
  isRecurring?: boolean;
  recurringPattern?: string;
  userId: number;
  createdAt?: string;
}
